conda/version.py

```diff
@@ -302,6 +302,10 @@ def __new__(cls, spec):
         self.spec = spec
         if isinstance(spec, tuple):
             self.match = self.all_match_ if spec[0] == 'all' else self.any_match_
+        elif spec.startswith("^") and spec.endswith("$"):
```

Comment:
Slick approach, and I kind of agree with the "why not if we can". I don't see any drawbacks right away either..